### ISSUE TITLE
oned segfaults if configuration contains mads with same name

### ISSUE THREAD

**Issue by paczerny** [OpenNebula/one#5801]
**Description**
Oned segfaults if `oned.conf` contains duplicate mads There is no error in the `oned.log`

**To Reproduce**
Set to `oned.conf`:
```
VM_MAD = [ NAME="dummy", SUNSTONE_NAME="Testing", EXECUTABLE="one_vmm_dummy", TYPE="xml" ]
VM_MAD = [ NAME="dummy", SUNSTONE_NAME="Can be different", EXECUTABLE="one_vmm_dummy", TYPE="xml" ]
```
Start oned

**Expected behavior**
Oned should report error to `oned.log` and exit

**Details**
 - Affected Component: oned
 - Version: 6.2.3, master

**Additional context**
The error may occur also for `TM_MAD_CONF`, `DS_MAD_CONF`, `MARKET_MAD_CONF`, `AUTH_MAD_CONF`, `VN_MAD_CONF`

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by paczerny** [comment]
In wasn't segfaulting, it just doesn't terminate correctly and it doesn't report failure.

[PR sources](https://github.com/OpenNebula/one-ee/pull/2396)

**Comment by paczerny** [comment]
Fixed in the scope of #6687
